runtime/tests/interpreter/container_mutation_test.go

```diff
@@ -258,6 +259,104 @@ func TestArrayMutation(t *testing.T) {
 
 		assert.Equal(t, sema.StringType, mutationError.ExpectedType)
 	})
+
+	t.Run("function array mutation", func(t *testing.T) {
+		t.Parallel()
+
+		standardLibraryFunctions :=
+			stdlib.StandardLibraryFunctions{
+				stdlib.LogFunction,
+			}
+
+		valueDeclarations := standardLibraryFunctions.ToSemaValueDeclarations()
+		values := standardLibraryFunctions.ToInterpreterValueDeclarations()
+
+		inter, err := parseCheckAndInterpretWithOptions(t, `
+                fun test() {
+                    let array: [AnyStruct] = [nil] as [((AnyStruct):Void)?]
+
+                    let x = 5
+                    array[0] =  log
+                }
+            `,
+			ParseCheckAndInterpretOptions{
+				CheckerOptions: []sema.Option{
+					sema.WithPredeclaredValues(valueDeclarations),
+				},
+				Options: []interpreter.Option{
+					interpreter.WithPredeclaredValues(values),
+				},
+			},
+		)
+
+		require.NoError(t, err)
+
+		// TODO: Shouldn't throw an error once dynamic subtyping for functions is implemented.
```

Comment:
Could you please create a follow-up issue for this TODO?

Reply:
Created: https://github.com/onflow/cadence/issues/1145